proto/addressmetadata.proto

```diff
@@ -1,39 +1,74 @@
 syntax = "proto3";
 package models;
 
-message Header {
-    string Name = 1;
-    string Value = 2;
+// Specifies the revocation method
+enum RevocationMethod {
+    /* If any of the pubkeys given in commitment_data[..n-1] signs
+    data commitment_data[n] then revocation is granted. */
+    BackupPubkey = 0;
+}
+
+// Message sent to server to revoke a key
+message Revocation {
+    // The revocation method
+    RevocationMethod method = 1;
+    // Data associated with the reveal phase
+    repeated bytes reveal_data = 2;
+    // New key to redirect to
+    bytes pub_key_hash = 3;
+    // Memo attached to revocation
+    string memo = 4;
 }
 
-// Row is an indidual piece of structured data provided by wallet authors.
-// The headers help wallets determine how to deserialize the data for their purposes.
+// A RevocationClause defines conditions under which the a key can be revoked.
+message RevocationClause {
+    // The revocation method
+    RevocationMethod method = 1;
+    // Data associated with commitment phase
+    repeated bytes commitment_data = 2;
```

Comment:
Perhaps this is too generic

Reply:
I don't understand this.  The commitments don't need to be published until the revocation is occurring.

Reply:
The naming here is "commitment" vs "reveal".

The commitment would be the data providing the conditions for revocation (which need to known by everyone prior) and the reveal (see inside the `Revocation` message) would be the data fulfilling the conditions which are published later during revocation.